staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -87,12 +87,18 @@ func WithAudit(handler http.Handler, requestContextMapper request.RequestContext
 			return
 		}
 
+		// send first event for non-read-only requests
+		ri, _ := request.RequestInfoFrom(ctx)
+		if !(ri != nil && ri.ReadOnly) {
```

Comment:
This sound like the policy then. But I agree to  discuss that as follow-up. Also this PR was meant as a base of discussion. We don't have to decide here.